include/debug/coredump.h

```diff
@@ -136,6 +150,9 @@ void coredump_buffer_output(uint8_t *buf, size_t buflen);
 int coredump_query(enum coredump_query_id query_id, void *arg);
 int coredump_cmd(enum coredump_cmd_id cmd_id, void *arg);
 
+void coredump_register_memory_region(struct coredump_mem_region_node_t *region);
+bool coredump_unregister_memory_region(struct coredump_mem_region_node_t *region);
```

Comment:
I think this should hide behind the newly introduced kconfig. So that it will generate errors during compilation if the kconfig is not enabled, instead of silently ignoring the calls.